gen/enum.go

```diff
@@ -110,19 +114,27 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 					return nil
 			<end ->
 				default:
-					<$val>, err := <$strconv>.ParseInt(<$s>, 10, 32)
+					<if checkErrOnUnknownEnum ->
+						return <$fmt>.Errorf("unknown enum value %q for %q", <$s>, "<$enumName>")
+					<else ->
+					<$val>, err := strconv.ParseInt(<$s>, 10, 32)
 					if err != nil {
 						return <$fmt>.Errorf("unknown enum value %q for %q: %v", <$s>, "<$enumName>", err)
 					}
 					*<$v> = <$enumName>(<$val>)
 					return nil
+					<end ->
 			}
 		}
 
 		// MarshalText encodes <$enumName> to text.
 		//
-		// If the enum value is recognized, its name is returned. Otherwise,
-		// its integer value is returned.
+		// If the enum value is recognized, its name is returned.
+		<if checkErrOnUnknownEnum ->
+		//Otherwise, error is thrown
```

Comment:
*nit*: space after `//` for consistency with the rest of the generated comments.

Reply:
This comment is still valid.